dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -182,6 +187,90 @@ private static void printStaticLinkageErrors(
     return jarPaths.build();
   }
 
+  /**
+   * Finds linkage errors in the input classpath and generates a static linkage check report.
+   */
+  StaticLinkageCheckReport findLinkageErrors() {
+    ImmutableList<Path> jarFilePaths = classDumper.getInputClasspath();
+
+    ImmutableMap.Builder<Path, SymbolTable> jarToSymbols = ImmutableMap.builder();
+    for (Path jarPath : jarFilePaths) {
+      jarToSymbols.put(jarPath, scanExternalSymbolTable(jarPath));
+    }
+
+    // Validate linkage error of each reference
+    ImmutableList.Builder<JarLinkageReport> jarLinkageReports = ImmutableList.builder();
```

Comment:
Again, you can do this in one step if you'd like:

```
jarToSymbols.entrySet().stream()
    .map(entry -> findInvalidReferences(entry.getKey(), entry.getValue()))
    .collect(toImmutableList());
```

Reply:
I'd keep this for-loop because `findInvalidReferences` (currently empty implementation) will have `throws IOException, ClassNotFoundException` as it reads files. Functions with checked exception are not good with lambda so I keep this as for-loop.

(Originally `scanSymbolReferencesInJar` and `findInvalidReferences` were used with stream.)